pkg/api/reconciler/claim-reconciler/reconiler.go

```diff
@@ -72,63 +75,64 @@ func (r *objectBucketClaimReconciler) Reconcile(request reconcile.Request) (reco
 
 	klog.V(util.DebugLogLvl).Infof("Reconciling object %q", request)
 
-	obc := &v1alpha1.ObjectBucketClaim{}
-	if err := r.client.Get(context.TODO(), request.NamespacedName, obc); err != nil {
-		return handleErr("error getting claim %q: %v", obc.Name, err)
+	obc, err := r.claimFromKey(request.NamespacedName)
+	if err != nil {
+		return handleErr("error getting claim for key %s: %v", err)
 	}
 
-	className := obc.Spec.StorageClassName
-
-	storageClass, err := util.GetStorageClassByName(className, r.client)
-	if err != nil && storageClass == nil {
-		return handleErr("unable to get storageClass %q of claim %q: %v", className, obc.Name, err)
+	if !r.shouldProvision(obc) {
+		return handleErr("skipping provisioning for claim %s", obc.Name)
 	}
 
-	if !r.shouldProvision(storageClass) {
-		return handleErr("claim failed shouldProvision check")
+	class, err := util.StorageClassForClaim(obc, r.client, r.ctx)
+	if err != nil {
+		return handleErr("unable to get storage class: %v", err)
 	}
 
-	reclaimPolicy, err := util.TranslateReclaimPolicy(*storageClass.ReclaimPolicy)
+	reclaimPolicy, err := util.TranslateReclaimPolicy(*class.ReclaimPolicy)
 	if err != nil {
-		return handleErr("error translating core.PersistentVolumeReclaimPolicy %q to v1alpha1.ReclaimPolicy: %v", storageClass.ReclaimPolicy, err)
+		return handleErr("error translating core.PersistentVolumeReclaimPolicy %q to v1alpha1.ReclaimPolicy: %v", class.ReclaimPolicy, err)
+	}
+
+	bucketName := obc.Spec.BucketName
```

Comment:
will the api validate (eventually) that bucketName and generateBucketName are mutually exclusive?